### PR TITLE
Move result.status to result.metadata.status as expected by PlaidLink…

### LINKED ISSUE #290: iOS tiggers onExit callback (with valid linked response) instead of onSuccess, Android works fine
# The problem

In the Lates version, I noticed the issue where on the Android, upon Success Linking I'll get a callback on the onSuccess method, but on the iOS onExit with onSuccess data.

Here is the response that's returned on the onExit method: 

```
{
  "metadata": {
    "accounts": [[Object]],
    "institution": {
      "id": "ins_3",
      "name": "Chase"
    },
    "linkSessionId": "2aae6b8f-45b2-4e0f-ab99-69bb7bb712bf",
    "metadataJson": "{\"institution\":{\"name\":\"Chase\",\"institution_id\":\"ins_3\"},\"account\":{\"id\":\"QlKDg8E681u7GPywA1rDHaqLJa7E3LhpPGnaa\",\"name\":\"Plaid Checking\",\"type\":\"depository\",\"subtype\":\"checking\",\"mask\":\"0000\"},\"account_id\":\"QlKDg8E681u7GPywA1rDHaqLJa7E3LhpPGnaa\",\"accounts\":[{\"id\":\"QlKDg8E681u7GPywA1rDHaqLJa7E3LhpPGnaa\",\"name\":\"Plaid Checking\",\"mask\":\"0000\",\"type\":\"depository\",\"subtype\":\"checking\"}],\"link_session_id\":\"2aae6b8f-45b2-4e0f-ab99-69bb7bb712bf\",\"public_token\":\"public-sandbox-fe3e2b27-73fd-4855-a476-49f107cb0fbd\"}"
  },
  "publicToken": "public-sandbox-fe3e2b27-73fd-4855-a476-49f107cb0fbd",
  "status": "connected"
}
```

**Problem Comes from this method:** 

https://github.com/plaid/react-native-plaid-link-sdk/blob/master/PlaidLink.tsx#L67

As you can see on the response above, "**status**" is not a part of the "**metadata**" object. Any idea why this structure is different on the iOS and Android? 

And is it possible that the problem is with my local setup or it comes from the Plaid API?, Thanks!

# Environment

|                        |  |
| ------------------------| ---------- |
| Plaid Link React Native          | 7.0.0-rc1                |
| ReactNative Version     | ~0.63.4                |
| Occurs on Android             | no                |
| Occurs on iOS     | yes                |
| iOS Version           | 11                |
| iOS Devices/Emulators               | iPhone 11                |

# Expected Result
onSuccess Triggered when Linking is done properly

### PR DESCRIPTION
**Pull request by pladaub**
….tsx

closes #290 290

### CODE REVIEW

**Comment by pladaub** [comment]
After speaking with @afh moved the change from being inside the ObjC bridge to being inside the JS layer

**Comment by amytang0** [comment]
This still returns in onExit for me

**Review by amytang0 — APPROVED** [review]
LGTM

File: PlaidLink.tsx [hunk 1]
```diff
@@ -64,16 +64,16 @@ export const openLink = async (props: PlaidLinkProps) => {
             props.onExit(data);
           }
         } else {
-          switch (result.metadata.status) {
+          switch (result.status) {
```
Comment (afh) [inline PlaidLink.tsx:67]: As chatted earlier it is probably be best to not rely on status, but use the type of result to determine whether `onSuccess` or `onExit` should be called.

Comment (amytang0) [reply PlaidLink.tsx:67]: Agreed, and now we have types!

**Review by amytang0 — CHANGES_REQUESTED** [review]
Sorry, didn't see the removal of LinkExitMetadata.status in LinkExit as well, that's a value we should keep in LinkExit, and remove from LinkSuccess

**Comment by afh** [comment]
Closing in favor of #294

**Pull request closed without merging**